File: 06_Wait_For_It/part2.py
import os

with open(os.path.join(os.path.dirname(os.path.abspath(__file__)), "input"), "r") as file:
    time, record = [int(line[9:].strip().replace(" ", "")) for line in file.readlines()]

ways = 0

for speed in range(1, time):
    distance = speed * (time - speed)

    if distance > record:
        ways += 1

print(f"The number of ways is {ways}")

File: 06_Wait_For_It/part2.js
const fs = require("node:fs");
const path = require("path");

const data = fs.readFileSync(path.resolve(__dirname, "./input"), "utf-8");
const [time, record] = data.trimEnd().split("\n").map(line => line.slice(9).replaceAll(" ", "")).map(Number);

let ways = 0;

for (let speed = 1; speed < time; speed++) {
  const distance = speed * (time - speed);

  if (distance > record) {
    ways += 1;
  }
}

console.log(`The number of ways is ${ways}`);
